Add change to VARCHAR in the upgrade guide (#16216)
apache/datafusion: docs/source/library-user-guide/upgrading.md

## DataFusion `48.0.0`

> ### `VARCHAR` SQL type changed to map to `Utf8View` Arrow type

Reviewer: ```suggestion
### The `VARCHAR` SQL type is now represented as `Utf8View` in Arrow.
```

This means that when you create a table with a `VARCHAR` column, it will now use
`Utf8View` as the underlying data type. For example:

```sql
> CREATE TABLE my_table (my_column VARCHAR);
0 row(s) fetched.
Elapsed 0.001 seconds.

> DESCRIBE my_table;
+-------------+-----------+-------------+
| column_name | data_type | is_nullable |
+-------------+-----------+-------------+
| my_column   | Utf8View  | YES         |
+-------------+-----------+-------------+
1 row(s) fetched.
Elapsed 0.000 seconds.
```

You can restore the old behavior of using `Utf8` by changing the
`datafusion.sql_parser.map_varchar_to_utf8view` configuration setting. For
example

```sql
> set datafusion.sql_parser.map_varchar_to_utf8view = false;
0 row(s) fetched.
Elapsed 0.001 seconds.

> CREATE TABLE my_table (my_column VARCHAR);
0 row(s) fetched.
Elapsed 0.014 seconds.

> DESCRIBE my_table;
+-------------+-----------+-------------+
| column_name | data_type | is_nullable |
+-------------+-----------+-------------+
| my_column   | Utf8      | YES         |
+-------------+-----------+-------------+
1 row(s) fetched.
Elapsed 0.004 seconds.
```

> 

Reviewer 2: Thank you @alamb , i think we also need to add note:

If customers use avro format, for datafusion 48.0, we need to set datafusion.sql_parser.map_varchar_to_utf8view = false;

Because the support for utf8view for avro are not releasing currently.

https://github.com/apache/arrow-rs/issues/7262

Author: Than you @zhuqi-lucas  -- good call. Hopefully by the time we release Datafusion 48 the upstream supprt for avro will be done too.
